File: code_learning/比赛/2024-12-15/B.py
import string

# O(|S| * 26)
s = input()
ans: int = 10 ** 9
for ch in string.ascii_lowercase:
    cnt: int = 0
    for i, si in enumerate(s):
        step = abs(ord(si) - ord(ch))
        cnt += min(step, 26 - step)
    ans = min(ans, cnt)
print(ans)

File: code_learning/比赛/2024-12-15/B.cpp
#include <bits/stdc++.h>
using i64 = long long;

void solve() {
    std::string s; std::cin >> s;
    i64 ans = LLONG_MAX, cnt = 0;

    for (char i = 'a'; i <= 'z'; i++) {
        cnt = 0;
        for (int j = 0, step; j < s.size(); j++) {
            step = ((s[j] - i) + 26) % 26;
            cnt += std::min(step, 26 - step);
        }
        ans = std::min(ans, cnt);
    }
    std::cout << ans << '\n';
}

int main() {
    std::cin.tie(nullptr)->sync_with_stdio(false);

    solve();

    return 0;
}